src/main/java/com/google/starfish/services/LabelService.java

```diff
@@ -46,6 +50,57 @@ public void insertMiscLabel(DataSource pool, String label) {
     insertLabel(pool, label, Type.MISC);
   }
 
+  /** Returns a map of all school and course labels in the database */
+  public HashMap<String, String[]> getAllSchoolAndCourseLabels(DataSource pool) {
+    HashMap<String, String[]> organizedLabels = new HashMap<>();
+    try (Connection conn = pool.getConnection()) {
+      List<String> schools = new ArrayList<>();
+      List<String> courses = new ArrayList<>();
+      try {
+        conn.setAutoCommit(false);
+        String stmt1 = 
+            "SELECT * " 
+          + "FROM " + LABELS + " "
+          + "WHERE type= \"" + Type.SCHOOL.getType() + "\"";
+        String stmt2 = 
+            "SELECT * " 
+          + "FROM " + LABELS + " "
+          + "WHERE type= \"" + Type.COURSE.getType() + "\"";
+        try (PreparedStatement schoolStmt = conn.prepareStatement(stmt1); 
+            PreparedStatement courseStmt = conn.prepareStatement(stmt2)) {
+          ResultSet schoolResults = schoolStmt.executeQuery();
+          ResultSet courseResults = courseStmt.executeQuery();
+          conn.commit();
+          while (schoolResults.next()) {
+            String school = schoolResults.getString("title");
+            schools.add(school);
+          }
+          while (courseResults.next()) {
+            String course = courseResults.getString("title");
+            courses.add(course);
+          }
+          organizedLabels.put("schools", schools.toArray(new String[0]));
+          organizedLabels.put("courses", courses.toArray(new String[0]));
```

Comment:
since you're doing the same thing for both schools and courses, there's probably a way you can extract this behavior out into a helper function and condense the logic in this try branch a bit more

Reply:
This is a good point, I'll make a function that takes a ResultSet and returns an Array of everything in the ResultSet; that should make this more readable.